service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -181,6 +183,8 @@ private String buildQuantityText(MedicationRequest medicationRequest) {
             return medicationRequest.getDispenseRequest().getQuantity().getUnit();
         } else if (medicationRequest.hasDispenseRequest() && medicationRequest.getDispenseRequest().hasExtension()) {
             return extractDispenseRequestQuantityText(medicationRequest);
+        } else if (medicationRequest.hasDispenseRequest() && medicationRequest.getDispenseRequest().getQuantity().hasExtension()) {
```

Comment:
Should you be checking whether this has a quantity? so getDispenseRequest().hasQuantity() ?

Reply:
Sure, fixed now